lib/testing/src/lib/core/actions/identity/query.service.ts

```diff
@@ -168,6 +169,46 @@ export class QueryService {
         return ApiUtil.waitForApi(apiCall, predicate);
     }
 
+    async getTask(taskName, processInstanceId, appName, status): Promise<any> {
+
+        const path = '/' + appName + '/query/v1/process-instances/' + processInstanceId + '/tasks';
+        const method = 'GET';
+
+        const queryParams = {}, postBody = {};
+
+        const data = await this.api.performBpmOperation(path, method, queryParams, postBody);
+        for (let i = 0; i < data.list.entries.length; i++) {
+            if (data.list.entries[i].entry.name === taskName) {
+                const task = data.list.entries[i];
+
+                if (task.entry.status === status) {
+                    return task;
+                } else if (this.retryCount > 0) {
+                    this.retryCount--;
+                    return this.getTask(taskName, processInstanceId, appName, status);
```

Comment:
must not share the state via the global variable, otherwise 2+ calls of this method will fail. Instead, pass the variable recursively
```suggestion
                    return this.getTask(taskName, processInstanceId, appName, status, retryCount--);
```

Reply:
Modified as per your suggestion